plugins/restapi/handlers.go

```diff
@@ -315,6 +326,123 @@ func (p *Plugin) registerHTTPHandler(key, method string, f func() (interface{},
 	p.HTTPHandlers.RegisterHTTPHandler(key, handlerFunc, method)
 }
 
+// jsonSchemaHandler returns JSON schema of VPP-Agent configuration.
+func (p *Plugin) jsonSchemaHandler(formatter *render.Render) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		// create FileDescriptorProto for dynamic Config holding all VPP-Agent configuration
+		knownModels, err := client.LocalClient.KnownModels("config") // locally registered models
+		if err != nil {
+			errMsg := fmt.Sprintf("500 Internal server error: "+
+				"can't get registered models: %v\n", err)
+			p.Log.Error(errMsg)
+			p.logError(formatter.JSON(w, http.StatusInternalServerError, errMsg))
+			return
+		}
+		config, err := client.NewDynamicConfig(knownModels)
+		if err != nil {
+			errMsg := fmt.Sprintf("500 Internal server error: "+
+				"can't create dynamic config due to: %v\n", err)
+			p.Log.Error(errMsg)
+			p.logError(formatter.JSON(w, http.StatusInternalServerError, errMsg))
+			return
+		}
+		dynConfigFileDescProto := protodesc.ToFileDescriptorProto(config.ProtoReflect().Descriptor().ParentFile())
+
+		// create list of all FileDescriptorProtos (imports should be before converted proto file -> dynConfig is last)
+		fileDescriptorProtos := allFileDescriptorProtos(knownModels)
+		fileDescriptorProtos = append(fileDescriptorProtos, dynConfigFileDescProto)
+
+		// creating input for protoc's plugin (code extracted in plugins/restapi/jsonschema) that can convert
+		// FileDescriptorProtos to JSONSchema
+		params := "messages=[Dynamic_config]" + // targeting only the main config message (proto file has also other messages)
+			",disallow_additional_properties" + // additional unknown json fields makes configuration applying fail
+			",proto_and_json_fieldnames" // allow also proto names for fields
+		cgReq := &protoc_plugin.CodeGeneratorRequest{
+			ProtoFile:       fileDescriptorProtos,
+			FileToGenerate:  []string{dynConfigFileDescProto.GetName()},
+			Parameter:       &params,
+			CompilerVersion: nil, // compiler version is not need in this protoc plugin
+		}
+		cgReqMarshalled, err := proto.Marshal(cgReq)
+		if err != nil {
+			fmt.Errorf("can't proto marshal to: %w", err)
```

Comment:
probably a typo that return value is not used

Reply:
It is forgotten leftover from error handling changes. Fixed in [`cd09327`](https://github.com/ligato/vpp-agent/pull/1784/commits/cd093279785a4b0ce46f06cc6b03e9d21bd12a76).